### PR TITLE
Revert "bpo-28533: Remove asyncore, asynchat, smtpd modules (GH-29521)"

### PR DESCRIPTION
**Pull request by vstinner**
This reverts commit 9bf2cbc4c498812e14f20d86acb61c53928a5a57.

https://bugs.python.org/issue28533

### CODE REVIEW

**Comment by vstinner** [comment]
PR related to the SC decision: https://github.com/python/steering-council/issues/86

File: Doc/library/socketserver.rst [hunk 1]
```diff
@@ -176,7 +176,7 @@ partially finished requests and to use :mod:`selectors` to decide which
 request to work on next (or whether to handle a new incoming request).  This is
 particularly important for stream services where each client can potentially be
 connected for a long time (if threads or subprocesses cannot be used).  See
-:mod:`asyncio` for another way to manage this.
+:mod:`asyncore` for another way to manage this.
```
Comment (warsaw) [inline Doc/library/socketserver.rst:179]: Can't you still recommend `asyncio` even with the revert?

**Review by warsaw — APPROVED** [review]

**Review by vsajip — APPROVED** [review]

**Comment by hroncok** [comment]
@pablogsal Can we please include this in 3.11.0a3?

**Comment by pablogsal** [comment]
> @pablogsal Can we please include this in 3.11.0a3?

Yup, but let's make sure It's merged today. We have a bunch of release blockers anyway :(

File: Doc/library/socketserver.rst (hunk 1 above)
Comment (vstinner) [reply Doc/library/socketserver.rst:179]: Yes, it's possible, but I wanted to use a simple "git revert" command. Moreover, Miro wants to get it in 3.11.0a3, and Pablo suggests to merge the PR today, so I simply merged my PR.

**Pull request merged**